drivers/espi/espi_mchp_xec.c

```diff
@@ -471,11 +469,13 @@ static int espi_xec_receive_oob(struct device *dev,
 	}
 
 	/* Check if buffer passed to driver can fit the received packet */
-	if (ESPI_OOB_REGS->RX_LEN > pckt.len) {
+	if ((ESPI_OOB_REGS->RX_LEN & ESPI_XEC_OOB_RX_LEN_MASK) > pckt.len) {
+		LOG_ERR("space rcvd %d vs %d",
+			ESPI_OOB_REGS->RX_LEN, pckt.len);
```

Comment:
Should this `RX_LEN` need to be masked too? Also please expand the log message to explain the error (maybe something like "received oob packet longer than buffer"?).

Reply:
Yes